experimental/examples/imageviewer/shared/src/iosMain/kotlin/example/imageviewer/storage/IosImageStorage.ios.kt

```diff
@@ -0,0 +1,180 @@
+            val byteArray: ByteArray = ByteArray(jpgRepresentation.length.toInt()).apply {
+                usePinned {
+                    memcpy(it.addressOf(0), jpgRepresentation.bytes, jpgRepresentation.length)
+                }
+            }
+            Image.makeFromEncoded(byteArray).toComposeImageBitmap()
+        }.await()
+
+    override suspend fun getImage(picture: PictureData.Camera): ImageBitmap =
+        ioScope.async {
+            fun getFileContent() = readPngFromFile(picture.pngFile)
+            var jpgRepresentation: NSData? = getFileContent()
+            while (jpgRepresentation == null) {
+                yield()
+                jpgRepresentation = getFileContent()
+            }
+            val byteArray: ByteArray = ByteArray(jpgRepresentation.length.toInt()).apply {
+                usePinned {
+                    memcpy(it.addressOf(0), jpgRepresentation.bytes, jpgRepresentation.length)
+                }
+            }
+            Image.makeFromEncoded(byteArray).toComposeImageBitmap()
+        }.await()
+
+}
+
+private fun UIImage.resizeToThumbnail(): UIImage {
+    val targetScale = maxOf(
+        storableThumbnailSize.toFloat() / size.useContents { width },
+        storableThumbnailSize.toFloat() / size.useContents { height },
+    )
+    val newSize = size.useContents { CGSizeMake(width * targetScale, height * targetScale) }
+    return resize(newSize)
+}
+
+private fun UIImage.resizeToBig(): UIImage {
+    val targetScale = maxOf(
+        maxStorableImageSize.toFloat() / size.useContents { width },
+        maxStorableImageSize.toFloat() / size.useContents { height },
+    )
+    val newSize = size.useContents { CGSizeMake(width * targetScale, height * targetScale) }
+    return resize(newSize)
+}
+
+private fun UIImage.resize(targetSize: CValue<CGSize>): UIImage {
+    val currentSize = this.size
+    val widthRatio = targetSize.useContents { width } / currentSize.useContents { width }
+    val heightRatio = targetSize.useContents { height } / currentSize.useContents { height }
+
+    val newSize: CValue<CGSize> = if (widthRatio > heightRatio) {
+        CGSizeMake(
+            width = currentSize.useContents { width } * heightRatio,
+            height = currentSize.useContents { height } * heightRatio
+        )
+    } else {
+        CGSizeMake(
+            width = currentSize.useContents { width } * widthRatio,
+            height = currentSize.useContents { height } * widthRatio
+        )
+    }
+    val newRect = CGRectMake(
+        x = 0.0,
+        y = 0.0,
+        width = newSize.useContents { width },
+        height = newSize.useContents { height }
+    )
+    UIGraphicsBeginImageContextWithOptions(size = newSize, opaque = false, scale = 1.0)
+    this.drawInRect(newRect)
+    val newImage = UIGraphicsGetImageFromCurrentImageContext()
+    UIGraphicsEndImageContext()
+
+    return newImage!!
+}
+
+private val PictureData.Camera.pngFile get():String = id + ".jpg"
```

Comment:
png file is jpg?
And fix formatting here please

Reply:
Renamed all to jpg